src/App.tsx

```diff
@@ -1,26 +1,171 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import * as todoService from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './Components/TodoList';
+import { Footer } from './Components/Footer';
+import { Header } from './Components/Header';
+import cn from 'classnames';
+import { ErrorTypes } from './types/ErrorTypes';
+import { Loader } from './Components/Loader/Loader';
+import { Status } from './types/Status';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [listOfTodos, setListOfTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedValue, setSelectedValue] = useState(Status.all);
+  const [errorMessage, setErrorMessage] = useState<ErrorTypes | null>(null);
+  const [query, setQuery] = useState('');
+  const [isResponding, setisResponding] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoId, setLoadingTodoId] = useState<number[]>([]);
+
+  useEffect(() => {
+    setIsLoading(true);
+    todoService
+      .getTodos()
+      .then(setListOfTodos)
+      .catch(() => {
+        setErrorMessage(ErrorTypes.UnableToLoad);
+        setTimeout(() => {
+          setErrorMessage(null);
+        }, 3000);
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  if (!todoService.USER_ID) {
     return <UserWarning />;
   }
 
+  const getFilteringTodos = listOfTodos.filter(todo => {
+    switch (selectedValue) {
+      case Status.active:
+        return !todo.completed;
+      case Status.completed:
+        return todo.completed;
+      default:
+        return true;
+    }
+  });
+
+  const handleCompleted = (n: number) => {
+    const updatedTodos = listOfTodos?.map(todo =>
+      todo.id === n ? { ...todo, completed: !todo.completed } : todo,
+    );
+
+    setListOfTodos(updatedTodos);
+  };
+
+  const addTodo = ({ userId, title, completed }: Omit<Todo, 'id'>) => {
+    setErrorMessage(null);
+    setTempTodo({ id: 0, userId, title, completed });
+    setisResponding(true);
+
+    return todoService
+      .addTodo({ userId, title, completed })
+      .then(newTodo => {
+        setListOfTodos(currentTodos => [...currentTodos, newTodo]);
+      })
+      .catch(err => {
+        setErrorMessage(ErrorTypes.UnableToAdd);
+        setTimeout(() => {
+          setErrorMessage(null);
+        }, 3000);
+        throw err;
```

Comment:
do not throw an error here - you handled it already, so no need to throw it again. It will lead to a page crash (because you don't try to catch the error thrown here somewhere else)